### PR TITLE
add accelerate sound dependant on speed

### PR DESCRIPTION
**Pull request by dj24**
Added a constant revving noise that has its speed and volume altered depending on the speed of the car and boost state.

### CODE REVIEW

File: src/models/vehicle/Vehicle.jsx [hunk 1]
```diff
@@ -114,16 +115,19 @@ function VehicleAudio() {
...
     speed = mutation.speed
     controls = useStore.getState().controls
 
     boostAudio.current.setVolume(sound ? (controls.boost ? Math.pow(speed / maxSpeed, 1.5) + 0.5 : 0) * 5 : 0)
     boostAudio.current.setPlaybackRate(Math.pow(speed / maxSpeed, 1.5) + 0.5)
-    engineAudio.current.setVolume(sound ? 1 : 0)
+    engineAudio.current.setVolume(sound ? 1 - (speed / maxSpeed) * 2 : 0)
     accelerateAudio.current.setVolume(sound ? (speed / maxSpeed) * 2 : 0)
+    rpmTarget = Math.pow(speed / maxSpeed, 1.5) + (controls.boost ? 0.55 : 0.5)
+    accelerateAudio.current.setPlaybackRate(THREE.MathUtils.lerp(accelerateAudio.current.playbackRate, rpmTarget, delta * 20))
```
Comment (Gusted) [inline src/models/vehicle/Vehicle.jsx:130]:
```suggestion
    accelerateAudio.current.setPlaybackRate(MathUtils.lerp(accelerateAudio.current.playbackRate, rpmTarget, delta * 20))
````
MathUtils is already being imported from Three as constant.

**Comment by drcmda** [comment]
@dj24 do you think we could also get that popping exhaust noise every now and then?

**Comment by dj24** [comment]
> @dj24 do you think we could also get that popping exhaust noise every now and then?

ooh nice idea, im thinking we figure how to get it to sound like gears first, then we can add the pop when changing gears? idk if im thinking of the right noise though

**Comment by drcmda** [comment]
speed / maxspeed is between 0-1 so we could divide it by 5 and then we have ranges

**Pull request merged**